### PR TITLE
boards: PHYTEC: remove SoC prefix from board name

### PR DESCRIPTION
**Pull request by jonas-rem**
For easier recognition, PHYTEC boards have been prefixed with the SoC name. As the new hardware model includes the SoC and cpu, prefixing is not needed anymore. All PHYTEC eval boards have an individual and unique name and can be found easily via this name.

### CODE REVIEW

File: boards/phytec/phyboard_polis/doc/index.rst [hunk 1]
```diff
@@ -1,4 +1,4 @@
-.. _mimx8mm_phyboard_polis:
+.. _phyboard_polis:
 
 PhyBOARD Polis (NXP i.MX8M Mini)
```
Comment (dnltz) [inline boards/phytec/phyboard_polis/doc/index.rst:3]: Can we fix both titles in this PR as well?

Should be called `phyBOARD-Polis (NXP i.MX8M Mini)` or `phyBOARD-Polis i.MX8M Mini`

File: boards/phytec/phyboard_pollux/doc/index.rst [hunk 2]
```diff
@@ -1,4 +1,4 @@
-.. _mimx8mp_phyboard_pollux:
+.. _phyboard_pollux:
 
 PhyBOARD Pollux (NXP i.MX8M Plus)
```
Comment (dnltz) [inline boards/phytec/phyboard_pollux/doc/index.rst:3]: Same here.

File: boards/phytec/phyboard_polis/doc/index.rst (hunk 1 above)
Comment (jonas-rem) [reply boards/phytec/phyboard_polis/doc/index.rst]: Just fixed all occasions of the name and corrected it for both boards. Should now be aligned over all PHYTEC boards.

**Review by nordicjm — CHANGES_REQUESTED** [review]
`boards/deprecated.cmake` needs update

**Comment by dnltz** [comment]
@jonas-rem  Here is a reference for `boards/deprecated.cmake`: https://github.com/zephyrproject-rtos/zephyr/blob/9f60075d14a283602a3580b6b97dd4686b53a28e/boards/deprecated.cmake#L49

**Review by kartben — CHANGES_REQUESTED** [review]
A welcome but breaking change so please mention it in the migration guide.
Also add redirect entries in doc/scripts/_redirects.py

Edit: possibly non breaking as others mentioned by using deprecated board name. Still needs a mention in the release notes to give people a heads-up though

**Comment by jonas-rem** [comment]
Thanks for the hints @nordicjm, @kartben, @dnltz!

Added an entry to `deprecated.yaml`, `redirect.py` and `release-notes-4.0.rst `

File: boards/phytec/phyboard_polis/doc/index.rst [hunk 3]
```diff
@@ -1,7 +1,7 @@
-.. _mimx8mm_phyboard_polis:
+.. _phyboard_polis:
 
-PhyBOARD Polis (NXP i.MX8M Mini)
-################################
+phyBOARD-Polis i.MX8M Mini
```
Comment (kartben) [inline boards/phytec/phyboard_polis/doc/index.rst:3]: One last request from me: please set the new names in the new `full_name` property of the respective `board.yml` files. Thanks!

**Review by kartben — CHANGES_REQUESTED** [review]

File: boards/phytec/phyboard_polis/doc/index.rst (hunk 3 above)
Comment (dnltz) [reply boards/phytec/phyboard_polis/doc/index.rst:3]: Ah nice, I will update the other boards soon too.

**Review by kartben — DISMISSED** [review]
looking good :) https://builds.zephyrproject.io/zephyr/pr/79575/docs/boards/index.html#vendor=phytec

(don't mind the bug in the vendor select box -- it will be fixed with #79695)

**Review by dnltz — DISMISSED** [review]
Thanks for updating those boards :)

**Comment by dnltz** [comment]
Assigned @carlescufi since he's the code owner.

File: boards/deprecated.cmake [hunk 4]
```diff
@@ -916,3 +916,6 @@ set(yd_esp32_DEPRECATED
 set(mimx8mp_phyboard_pollux/mimx8ml8/m7_DEPRECATED
     phyboard_pollux/mimx8ml8/m7
 )
+set(mimx8mm_phyboard_polis/mimx8mm6/m4_DEPRECATED
```
Comment (nordicjm) [inline boards/deprecated.cmake:919]: line 388 needs updating too

File: boards/phytec/phyboard_polis/doc/index.rst [hunk 5]
```diff
@@ -70,8 +70,8 @@ More information about the board can be found at the
 Supported Features
 ==================
 
-The Zephyr ``mimx8mm_phyboard_polis/mimx8mm6/m4`` board configuration supports the following
-hardware features:
+The Zephyr ``phyboard_polis/mimx8mm6/m4`` board configuration supports the
```
Comment (nordicjm) [inline boards/phytec/phyboard_polis/doc/index.rst:73]: *board target

File: boards/deprecated.cmake (hunk 4 above)
Comment (jonas-rem) [reply boards/deprecated.cmake:919]: Oh, right! The old deprecation that has been introduced when HWMv2 came must directly point to the most recent board. Will change.

Changed also l.368 (Polis).

File: boards/phytec/phyboard_polis/doc/index.rst (hunk 5 above)
Comment (jonas-rem) [reply boards/phytec/phyboard_polis/doc/index.rst]: fixed

**Comment by jonas-rem** [comment]
Adressed the last two comments from @nordicjm.

**Review by kartben — DISMISSED** [review]

**Review by nordicjm — DISMISSED** [review]

**Review by dnltz — DISMISSED** [review]

**Comment by jonas-rem** [comment]
I had to rebase on current main because of merge conflicts in the release notes file. There are no code changes introduced.

@kartben @nordicjm @dnltz @carlescufi could you please re-approve?

**Review by dnltz — DISMISSED** [review]

**Review by nordicjm — DISMISSED** [review]

**Review by carlescufi — DISMISSED** [review]

**Comment by MaureenHelm** [comment]
Needs rebase

**Comment by jonas-rem** [comment]
Rebased on current main

**Review by dnltz — APPROVED** [review]

**Review by kartben — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**